sdk/loadtestservice/load-testing-rest/src/beginFileValidation.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { FileUploadAndValidatePoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient } from "./index.js";
+import { TestUploadFile201Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { ReadStream } from "fs";
+
+/**
+ * Submits a Notification Hub job and creates a poller to poll for results.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginFileValidation(
+  client : AzureLoadTestingClient,
+  testId: string,
+  fileName: string,
+  file: ReadStream,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<FileUploadAndValidatePoller> {
+  const fileUploadResult = await client
+    .path("/tests/{testId}/files/{fileName}", testId, fileName)
+    .put({
+      contentType: "application/octet-stream",
+      body: file,
+    });
+
+  if (isUnexpected(fileUploadResult)) {
+    throw fileUploadResult.body.error;
+  }
```

Comment:
Could this be moved outside the `beginFileValidation` function? such that `beginFileValidation` takes the `fileUploadResult` as a parameter.

Reply:
Discussed in Arch Board meeting that both the operation, upload and validate should be in the same function as part of LRO.